src/plugins/workspace/public/components/data_source_association/data_source_association.tsx

```diff
@@ -68,6 +77,14 @@ export const DataSourceAssociation = ({ excludedDataSourceIds, onComplete, onErr
 
           if (res.success) {
             failedCount = res.result.filter((r) => !!r.error).length;
+            if (!defaultDataSourceId && uiSettings) {
+              // If a default data source is not already set, automatically set the first selected data source as default.
+              await uiSettings.set(
+                DEFAULT_DATA_SOURCE_UI_SETTINGS_ID,
+                objects[0].id,
+                UiSettingScope.WORKSPACE
+              );
+            }
```

Comment:
Seems we will trigger onComplete callback, can we set default data source there?